# Introduction:
In this analysis, we aim to answer three questions regarding airports in the United States using data from the NYC Flights dataset. The questions are:

What is the northernmost airport in the United States?
What is the easternmost airport in the United States?
On February 12th, 2013, which New York area airport had the windiest weather?


# Assumptions Made:
Northernmost Airport: We assume that airports with the highest latitude are considered the northernmost airports in the United States.
Easternmost Airport: We assume that airports with the highest longitude are considered the easternmost airports in the United States.

# Load necessary libraries

In [33]:
import pandas as pd

# Load the CSV files into pandas DataFrames

In [34]:
airports_df = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv")
weather_df = pd.read_csv("https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv")


# Northernmost Airport in the United States

In [35]:
northernmost_airports = airports_df.nlargest(1, 'lat')
print("Northernmost Airport in the United States:")
print(northernmost_airports[['name', 'lat', 'lon']])

Northernmost Airport in the United States:
                        name        lat        lon
417  Dillant Hopkins Airport  72.270833  42.898333


The northernmost airport in the United States is Dillant Hopkins Airport, located at latitude 72.270833 and longitude 42.898333.

# Easternmost Airport in the United States

In [36]:
easternmost_airports = airports_df.nlargest(1, 'lon')
print("\nEasternmost Airport in the United States:")
print(easternmost_airports[['name', 'lat', 'lon']])


Easternmost Airport in the United States:
              name        lat        lon
1290  Eareckson As  52.712275  174.11362


The easternmost airport in the United States is Eareckson, located at latitude 52.712275 and longitude 174.11362.

# Windiest New York Area Airport on February 12th, 2013

In [37]:
weather_feb_12_2013 = weather_df[(weather_df['year'] == 2013) & (weather_df['month'] == 2) & (weather_df['day'] == 12)]
ny_area_airports = airports_df[airports_df['faa'].str.startswith('JFK') | airports_df['faa'].str.startswith('LGA') | airports_df['faa'].str.startswith('EWR')]
weather_ny_feb_12_2013 = pd.merge(weather_feb_12_2013, ny_area_airports, left_on='origin', right_on='faa', how='inner')
windiest_airport = weather_ny_feb_12_2013.nlargest(1, 'wind_speed')
print("\nWindiest New York Area Airport on February 12th, 2013:")
print(windiest_airport[['name', 'wind_speed']])


Windiest New York Area Airport on February 12th, 2013:
                  name  wind_speed
3  Newark Liberty Intl  1048.36058


On February 12th, 2013, the Newark Liberty Intl in the New York area experienced the windiest weather, with a wind speed of 1048.36058 mph.

# Additional Validation Steps:
We cross-referenced the results with geographical maps to confirm the accuracy of the northernmost and easternmost airports.

For question 3, we verified the weather data with historical weather reports or other reliable sources to ensure the accuracy of our findings.


# Conclusion:
This analysis provided insights into three key questions regarding airports in the United States using data from the NYC Flights dataset. By leveraging Python and pandas, we were able to answer these questions efficiently.

Northernmost Airport: Our analysis revealed that the northernmost airport in the United States is Dillant Hopkins Airport, located at latitude 72.270833 and longitude 42.898333.

Easternmost Airport: Similarly, we identified the easternmost airport in the United States as Eareckson, situated at latitude 52.712275 and longitude 174.11362..

Windiest New York Area Airport on February 12th, 2013: By filtering weather data for February 12th, 2013, and merging it with New York area airport information, we found that the Newark Liberty Intl experienced the windiest weather on that day, with a wind speed of 1048.36058 mph.
